managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1466,6 +1452,20 @@ public void operationFailed(MetaStoreException e) {
 
                     metadataMutex.unlock();
 
+                    if (e instanceof BadVersionException) {
```

Comment:
Do we need to put this logic before releasing the lock? Now we changed the previous behavior. I'm not sure if it will cause other thread safety issues

Reply:
+1

Reply:
Addressed.